File: frontend/src/types/locations.ts
export type AccessibilityStatus = 'supported' | 'not_supported' | 'unknown';
export type ConfidenceLevel = 'high' | 'medium' | 'low' | 'unknown';

export interface LocationSummary {
  id: string;
  name: string;
  type: string;
  lat?: number | null;
  lon?: number | null;
  accessibility_status: AccessibilityStatus;
  confidence: ConfidenceLevel;
  note?: string | null;
  /** Line short codes (KJL, MRL, ...) populated for rail_station rows. */
  routes?: string[];
}

export interface LocationDetail extends LocationSummary {
  routes: string[];
  known_facilities: string[];
  source_list: string[];
  /** Official station amenities from mrt.com.my scrape (subset of hubs). */
  station_facilities?: string[];
  station_address?: string | null;
  station_hours_summary?: string | null;
  facility_source_url?: string | null;
}

export interface PlaceSelection {
  // Lightweight location shape used across planning flow and API requests.
  displayName: string;
  lat?: number | null;
  lon?: number | null;
  googlePlaceId?: string | null;
}

File: backend/app/schemas/locations.py
from typing import Literal

from pydantic import BaseModel


AccessibilityStatus = Literal["supported", "not_supported", "unknown"]


class LocationSummary(BaseModel):
    id: str
    name: str
    type: str = "rail_station"
    lat: float | None = None
    lon: float | None = None
    accessibility_status: AccessibilityStatus = "unknown"
    confidence: Literal["high", "medium", "low", "unknown"] = "unknown"
    note: str | None = None
    routes: list[str] = []


class LocationDetail(LocationSummary):
    # Detail view extends summary with station-specific enrichment fields.
    routes: list[str] = []
    known_facilities: list[str] = []
    source_list: list[str] = []
    # Scraped from mrt.com.my (CSV); optional enrichment for selected hubs.
    station_facilities: list[str] = []
    station_address: str | None = None
    station_hours_summary: str | None = None
    facility_source_url: str | None = None
